.cursor/rules/05-development-guidelines.mdc

```diff
@@ -0,0 +1,48 @@
+---
+description:
+globs:
+alwaysApply: false
+---
+# Pinecone Development Guidelines
+
+## Code Style
+
+This project follows standard Haskell code conventions:
+- Use 4-space indentation
+- Use camelCase for function and variable names
+- Use PascalCase for types and constructors
+- Use snake_case for record fields to match Pinecone API naming
+
+## Language Extensions
+
+The project enables several language extensions by default in the cabal file:
+- BlockArguments
+- DataKinds
+- DeriveGeneric
+- DeriveAnyClass
+- DerivingStrategies
+- DuplicateRecordFields
+- GeneralizedNewtypeDeriving
+- NamedFieldPuns
+- OverloadedLists
+- OverloadedStrings
+- RecordWildCards
+- TypeApplications
+- TypeOperators
+
+## Record Fields
+
+Many record fields begin with underscore (`_`) to avoid name clashes due to the DuplicateRecordFields extension.
```

Comment:
```suggestion
Some record fields begin with underscore (`_`) to match the Pinecone API field names (e.g. `_id` and `_score`, which have underscores in the Pinecone API).  Other record field names end with an underscore to conflicting with Haskell keywords (e.g. `data_` instead of `data`).
```